llvm/lib/Transforms/Scalar/ConstraintElimination.cpp

```diff
@@ -1460,6 +1460,13 @@ static bool checkAndReplaceCondition(
           checkCondition(Cmp->getPredicate(), Cmp->getOperand(0),
                          Cmp->getOperand(1), Cmp, Info))
     return ReplaceCmpWithConstant(Cmp, *ImpliedCondition);
+
+  if (Cmp->hasSameSign() && Cmp->isUnsigned())
```

Comment:
Might be good to add a brief comment here that we also try the signed predicate for samesign unsigned predicates.